llvm/lib/Target/AMDGPU/AMDGPUCallingConv.td

```diff
@@ -91,7 +91,11 @@ def CSR_AMDGPU_AGPRs : CalleeSavedRegs<
 >;
 
 def CSR_AMDGPU_SGPRs : CalleeSavedRegs<
-  (sequence "SGPR%u", 30, 105)
+  (add (sequence "SGPR%u", 30, 37),
+       (sequence "SGPR%u", 46, 53),
+       (sequence "SGPR%u", 62, 69),
+       (sequence "SGPR%u", 78, 85),
+       (sequence "SGPR%u", 94, 105))
```

Comment:
This isn't quite 8, maybe adjust the description to be more specific? Also should comment the scheme here. Also, I can never remember what level of SGPRs corresponds to what occupancy

Reply:
Yeah, I was thinking that as well. Maybe I can start with s32 and leave s30-s31 as caller saved?

Reply:
However, s30-s31 are for return address. [A patch](https://reviews.llvm.org/D114652) made them callee saved years ago. I can start from s40 though. However, s32, s33, as well as s34 are used for stack pointer, frame pointer, and base pointer, respectively. Is it a good idea to make them caller saved?

Reply:
No. Do not change the behavior of these special registers s30 - s34. Yes, start making the CSR partition from s40.

Reply:
We should be moving the special inputs to all be in callee saved sgprs. It's most critical that the return address be a callee save